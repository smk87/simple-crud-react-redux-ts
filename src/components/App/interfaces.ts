import { User } from "../../store/user/types";

// Define prop types for this component
export interface AppProps {}

export interface LinkDispatchProps {
  startAddUser: (user: User) => void;
  startDeleteUser: (user: User) => void;
}

export interface LinkStateProps {
  users: User[];
}

// Define state property types for this component
export interface AppState {
  [key: string]: string;
}
