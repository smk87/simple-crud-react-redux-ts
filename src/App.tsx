import React, { Component } from "react";
import "./App.css";
import Input from "./Input";

export type InputHandlerParam = React.ChangeEvent<HTMLInputElement>;

interface AppProps {}

interface AppState {
  [key: string]: string;
}

export class App extends Component<AppProps, AppState> {
  state = {
    name: "",
    age: ""
  };

  handleInput = (hEvent: InputHandlerParam): void => {
    this.setState({ [hEvent.target.name]: hEvent.target.value }, () => {
      console.log(this.state);
    });
  };

  render() {
    return (
      <div className="App">
        <h1>Simple CRUD App</h1>
        <br />
        <br />
        <Input type="text" label="name" inputHandler={this.handleInput} />
        <Input type="number" label="age" inputHandler={this.handleInput} />
      </div>
    );
  }
}

export default App;
