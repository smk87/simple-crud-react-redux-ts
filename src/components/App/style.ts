import css from "styled-jsx/css";

export default css`
  .App {
    h1 {
      color: blue;
    }

    .info {
      color: red;
    }
  }
`;
