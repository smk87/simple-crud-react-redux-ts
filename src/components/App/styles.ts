import css from "styled-jsx/css";

export default css`
  .App {
    h1 {
      color: darkblue;
    }

    .info {
      color: darkred;
    }
  }
`;
