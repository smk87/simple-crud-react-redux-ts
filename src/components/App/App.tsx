import React, { Component } from "react";
import { connect } from "react-redux";
import { ThunkDispatch } from "redux-thunk";
import { bindActionCreators } from "redux";
import { startAddUser } from "../../store/user/actions";
import { UserActionsTypes } from "../../store/user/types";
import { InputHandlerParam, ClickHandlerParam } from "../../types";
import {
  AppProps,
  AppState,
  LinkDispatchProps,
  LinkStateProps
} from "./interfaces";
import Input from "../Input/Input";
import UserList from "../UserList/UserList";
import { ReduxStoreState } from "../../store";
import "../../App.css";

// Combine All types of props
type Props = AppProps & LinkDispatchProps & LinkStateProps;

class App extends Component<Props, AppState> {
  state = {
    name: "",
    age: ""
  };

  handleInput = (hEvent: InputHandlerParam): void => {
    this.setState({ [hEvent.target.name]: hEvent.target.value });
  };

  handleAdd = (hEvent: ClickHandlerParam): void => {
    this.props.startAddUser({
      name: this.state.name,
      age: this.state.age
    });
  };

  render() {
    return (
      <div className="App">
        <h1>Simple CRUD App with React-Redux-TS</h1>
        <br />
        <br />
        <Input type="text" label="name" inputHandler={this.handleInput} />
        <Input type="number" label="age" inputHandler={this.handleInput} />
        <button onClick={this.handleAdd}>Add</button>
        <br />
        <br />
        <UserList users={this.props.users} />
      </div>
    );
  }
}

const mapStateToProps = (
  state: ReduxStoreState,
  ownProps: AppProps
): LinkStateProps => ({
  users: state.user.users
});

const mapDispatchToProps = (
  dispatch: ThunkDispatch<any, any, UserActionsTypes>,
  ownProps: AppProps
): LinkDispatchProps => ({
  startAddUser: bindActionCreators(startAddUser, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
