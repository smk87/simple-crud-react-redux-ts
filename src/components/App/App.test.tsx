import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "../../store";

// Smoke test
it("Smoke Test: renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={configureStore()}>
      <App />
    </Provider>,
    div
  );
});

// Snapshot test
it("Snapshot Test: renders without crashing", () => {
  const tree = renderer
    .create(
      <Provider store={configureStore()}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
