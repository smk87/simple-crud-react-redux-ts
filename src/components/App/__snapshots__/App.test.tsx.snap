// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test: renders without crashing 1`] = `
<div
  className="App"
>
  <h1>
    Simple CRUD App with React-Redux-TS
  </h1>
  <br />
  <br />
  <h4
    className="info"
  >
    Enter User Info
  </h4>
  <div
    className="input-base"
  >
    <p>
      name
    </p>
    <input
      name="name"
      onChange={[Function]}
      type="text"
    />
    <style>
      
    </style>
  </div>
  <div
    className="input-base"
  >
    <p>
      age
    </p>
    <input
      name="age"
      onChange={[Function]}
      type="number"
    />
    <style>
      
    </style>
  </div>
  <button
    onClick={[Function]}
  >
    Add
  </button>
  <br />
  <br />
  <style>
    
  </style>
</div>
`;
